### PR TITLE
Python 3.8 build and fix for ld

### PR DESCRIPTION
**Pull request by ashwinvis**
See #73

### CODE REVIEW

**Comment by ashwinvis** [comment]
To be honest this would be nothing more than a band aid. It would be better to take the approach in #69

**Comment by KOLANICH** [comment]
They are ignoring PRs. It is completely inacceptable. It seems that the repo needs a new maintainer and a mechanism to strip maintainership from a maintainer not doing its job.

**Comment by ashwinvis** [comment]
There has to be plausible reason, who knows :shrug:! More maintainers is a good idea if the current ones are unavailable. Tagging @johanneskoester (maintainer of snakemake which depends on datrie).

**Comment by frankier** [comment]
It's not unacceptable. (Although I understand getting impatient.) Either wait or fork. If it's blocking you somehow, vendorise/pull from github/work around it. I'd recommend reading this for a look from the other side: https://blog.burntsushi.net/foss/

**Comment by frankier** [comment]
By the way if your workflow needs installing from pypi, you can always push to pypi under another package name (I know it's not great but...).

**Comment by ashwinvis** [comment]
I think `pip install -e git+https://path/to/repo.git` or `pip install htttps://path/to/git-tagged-release.zip` is a better option rather than adding noise to PyPI. Certainly not optimal, just duplication of effort.

**Comment by KOLANICH** [comment]
>It's not unacceptable.

It is. It is disrespect to everyone sending PRs to ignore them. If they don't maintain the project, they should `archive` it to make it clear that PRs are not accepted and that we shouldn't expect anything.

The only reasonable exclusions are forces majeure like death, hospitalization, disablement, loss of sanity, inprisonment, hardware failure, war. But I saw at least 1 commit since I have sent my PR. This is not a commit reporting of project being closed. It Is just a regular commit. This just means that the maintainers don't respect us.

about burntsushi's blog post:

>So what does this have to do with FOSS? The key, for me anyway, was being able to put up a boundary between myself and unattended issues and pull requests. I had to find a way to say to myself: “I am volunteering my time and it is okay if I don't respond in a timely manner. I trust that most other people will understand this and be reasonable about it.”

It is OK to ignore issues, because issues is when someone wants you to do something for him for free. If te issue is really severe the one ho need it to be solved the most will implement and send a PR.

PRs is another situation. Someone has done something probably valuable for your project for free and expects you to merge it fast. It is just disrespect to ignore that. Especially if the patch is a few lines of non-complex code and can be merged by hitting a single button on GitHub.

>By the way if your workflow needs installing from pypi, you can always push to pypi under another package name (I know it's not great but...).

I usually use PEP 508 to specify direct links to dependencies.

**Comment by Artoria2e5** [comment]
You can maybe update the CI file to test 3.8 too, or even better just test `nightly` so we know what will break in the future. (people are asking for a key pointing to the latest stable at travis-ci/travis-ci#9180, but i don't think they are doing it anytime soon.)

**Comment by ashwinvis** [comment]
Done. Ideally I would get rid of Python 2 support also, but I would rather let the maintainers do it.

File: src/cdatrie.c [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/* Generated by Cython 0.29.11 */
+/* Generated by Cython 0.29.14 */
```
Comment (KOLANICH) [inline src/cdatrie.c]: We should remove cython-generated files and just regenerate them on each build.

Comment (ashwinvis) [reply src/cdatrie.c]: I agree, [I mentioned the same above](https://github.com/pytries/datrie/pull/76#issuecomment-574286987).

**Comment by johanneskoester** [comment]
@ashwinvis do you see a chance for this to end up in a release sooner or later?

**Comment by ashwinvis** [comment]
I am not the maintainer, just a humble user of Snakemake... I would urge you to send an email to the datrie maintainers and resolve this deadlock

File: src/cdatrie.c (hunk 1 above)
Comment (ashwinvis) [reply src/cdatrie.c]: I did not remove it, but I added cythonize support in #78

**Comment by tacaswell** [comment]
@KOLANICH Your comments here are out of line.

> Someone has done something probably valuable for your project for free and expects you to merge it fast. It is just disrespect to ignore that. 

I'm sorry you have this expectation, but that is not the reality (I suggest reading
https://snarky.ca/setting-expectations-for-open-source-participation/ for more context).  Your comments are far more disrespectful.

**Comment by ashwinvis** [comment]
I agree, @KOLANICH's remarks could have been spelled out better, but there is no denying that the maintainers *are* active on GitHub, yet they choose to ignore all these PRs for several months now. It is a completely unnecessary trouble for all to start with.

**Comment by tacaswell** [comment]
These commits were merged in #80 .  Thank you @ashwinvis

**Pull request closed without merging**